partiql-eval/src/main/kotlin/org/partiql/eval/internal/helpers/RecordUtility.kt

```diff
@@ -1,12 +1,17 @@
 package org.partiql.eval.internal.helpers
 
 import org.partiql.eval.internal.Record
-import org.partiql.value.PartiQLValue
-import org.partiql.value.PartiQLValueExperimental
+import org.partiql.eval.value.Datum
 
 internal object RecordUtility {
-    @OptIn(PartiQLValueExperimental::class)
-    fun Record.toPartiQLValueList(): List<PartiQLValue> = List(this.values.size) {
-        this.values[it].toPartiQLValue()
+    /**
+     * Converts the [Record] into an array of Datum, while coercing any missing values into null values.
+     */
+    fun Record.toDatumArrayCoerceMissing(): Array<Datum> = Array(this.values.size) {
+        val d = this@toDatumArrayCoerceMissing.values[it]
+        when (d.isMissing) {
+            true -> Datum.nullValue()
+            else -> d
+        }
     }
```

Comment:
Just clarifying why you're doing this: even though the comparator treats null/missing as equal, it isn't deterministic which value will be returned in the TreeSet -- so you need to coerce them ahead of time.

Also, I'm leaving a suggestion here because I don't think you need to create a new Array and move all previous pointers to the new one.

```suggestion
    fun Array<Datum>.coerceMissing() {
        for (i in 0 until this.size) {
            val d = this[i]
            if (d.isMissing) {
                this[i] = Datum.nullValue()
            }
        }
    }
```

Reply:
> Just clarifying why you're doing this: even though the comparator treats null/missing as equal, it isn't deterministic which value will be returned in the TreeSet -- so you need to coerce them ahead of time.

Yes, that's correct. I added a brief comment in the latest commit to give some more context on the behavior.

---
> Also, I'm leaving a suggestion here because I don't think you need to create a new Array and move all previous pointers to the new one.

Thanks for the suggestion. You're right -- it can be simplified quite a bit. I applied your suggestion in the latest commit.